src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/matchtoterm/TermFieldRewriter.java

```diff
@@ -213,15 +214,25 @@ public boolean isValidIdentifierForTerm(SQLIdentifierExpr expr) {
          */
         return
             !expr.getName().startsWith("_")
-            && (
-                (   expr.getParent() instanceof SQLBinaryOpExpr
-                    && ((SQLBinaryOpExpr) expr.getParent()).getOperator() == SQLBinaryOperator.Equality
-                )
-                || expr.getParent() instanceof SQLInListExpr
-                || expr.getParent() instanceof SQLInSubQueryExpr
-                || expr.getParent() instanceof SQLSelectOrderByItem
-                || expr.getParent() instanceof MySqlSelectGroupByExpr
-            );
+            && (isValidIdentifier(expr) || checkIfNestedIdentifier(expr));
+    }
+
+    private boolean checkIfNestedIdentifier(SQLIdentifierExpr expr) {
+        return
+            expr.getParent() instanceof SQLMethodInvokeExpr
+            && ((SQLMethodInvokeExpr) expr.getParent()).getMethodName().equals("nested")
+            && isValidIdentifier((SQLMethodInvokeExpr) expr.getParent());
```

Comment:
I assume that SQLMethodInvokeExpr is child of SQLExpr. Do we need the cast here?

Reply:
The last revision has cast removed, don't know why it is showing stale code.